submission/loglist_manager.go

```diff
@@ -52,14 +52,20 @@ func NewLogListManager(llr LogListRefresher) *LogListManager {
 func (llm *LogListManager) Run(ctx context.Context, llRefresh time.Duration) {
 	llm.llRefreshInterval = llRefresh
 	go schedule.Every(ctx, llm.llRefreshInterval, func(ctx context.Context) {
-		if ll, err := llm.RefreshLogList(ctx); err != nil {
-			llm.Errors <- err
-		} else if ll != nil {
-			llm.LLUpdates <- llm.ProduceClientLogList()
-		}
+		llm.RefreshLogListAndNotify(ctx)
 	})
 }
 
+// RefreshLogListAndNotify runs single Log-list refresh and propagates data and
+// errors to corresponding channels
+func (llm *LogListManager) RefreshLogListAndNotify(ctx context.Context) {
+	if ll, err := llm.RefreshLogList(ctx); err != nil {
+		llm.Errors <- err
+	} else if ll != nil {
```

Comment:
Is it possible for RefreshLogList() to return `(nil, nil)`?

Reply:
no, but should we rely onto this knowledge?

Reply:
It's reasonable to rely on the contract of a method, and if it shouldn't return `(nil, nil)`, then you generally don't need to code defensively against that (otherwise we'd have a lot of extra checks everywhere).

Reply:
Updated